sdk/core/core/test/cmocka/test_az_span.c

```diff
@@ -751,6 +751,98 @@ static void az_span_u32toa_overflow_fails(void** state)
   assert_true(az_span_u32toa(buffer, v, &out_span) == AZ_ERROR_INSUFFICIENT_SPAN_SIZE);
 }
 
+static void az_span_atod_test(void** state)
+{
+  (void)state;
+  uint8_t buff[10];
+  az_span buff_span = AZ_SPAN_FROM_BUFFER(buff);
+  az_span remainder = { 0 };
+  remainder = az_span_copy_u8(buff_span, '-');
+  remainder = az_span_copy_u8(remainder, '1');
+  remainder = az_span_copy_u8(remainder, '2');
+  remainder = az_span_copy_u8(remainder, '2');
+  remainder = az_span_copy_u8(remainder, '2');
+  remainder = az_span_copy_u8(remainder, '.');
+  remainder = az_span_copy_u8(remainder, '5');
+  remainder = az_span_copy_u8(remainder, '6');
+  remainder = az_span_copy_u8(remainder, '8');
+  remainder = az_span_copy_u8(remainder, '1');
+  double number = -1222.5681;
+
+  uint8_t convertion_buff[11];
+  az_span conv_span = AZ_SPAN_FROM_BUFFER(convertion_buff);
+
+  double result = 0;
+  assert_return_code(az_span_atod(buff_span, &conv_span, &result), AZ_OK);
+
+  assert_true(number - result < 0.00001);
+}
+
+static void az_span_atod_test_possitive(void** state)
+{
+  (void)state;
+  uint8_t buff[10];
+  az_span buff_span = AZ_SPAN_FROM_BUFFER(buff);
+  az_span remainder = { 0 };
+  remainder = az_span_copy_u8(buff_span, '9');
+  remainder = az_span_copy_u8(remainder, '1');
+  remainder = az_span_copy_u8(remainder, '2');
+  remainder = az_span_copy_u8(remainder, '2');
+  remainder = az_span_copy_u8(remainder, '2');
+  remainder = az_span_copy_u8(remainder, '.');
+  remainder = az_span_copy_u8(remainder, '5');
+  remainder = az_span_copy_u8(remainder, '6');
+  remainder = az_span_copy_u8(remainder, '8');
+  remainder = az_span_copy_u8(remainder, '1');
+  double number = 91222.5681;
+
+  uint8_t convertion_buff[11];
+  az_span conv_span = AZ_SPAN_FROM_BUFFER(convertion_buff);
+
+  double result = 0;
+  assert_return_code(az_span_atod(buff_span, &conv_span, &result), AZ_OK);
+
+  assert_true(number - result < 0.00001);
+}
+
+static void az_span_atod_test_not_enough_space(void** state)
+{
+  (void)state;
+  uint8_t buff[10];
+  az_span buff_span = AZ_SPAN_FROM_BUFFER(buff);
+  az_span remainder = { 0 };
+  remainder = az_span_copy_u8(buff_span, '9');
+  remainder = az_span_copy_u8(remainder, '1');
+  remainder = az_span_copy_u8(remainder, '2');
+  remainder = az_span_copy_u8(remainder, '2');
+  remainder = az_span_copy_u8(remainder, '2');
+  remainder = az_span_copy_u8(remainder, '.');
+  remainder = az_span_copy_u8(remainder, '5');
+  remainder = az_span_copy_u8(remainder, '6');
+  remainder = az_span_copy_u8(remainder, '8');
+  remainder = az_span_copy_u8(remainder, '1');
+
+  uint8_t convertion_buff[10];
+  az_span conv_span = AZ_SPAN_FROM_BUFFER(convertion_buff);
+
+  double result = 0;
+  az_result result_op = az_span_atod(buff_span, &conv_span, &result);
+  assert_true(result_op == AZ_ERROR_INSUFFICIENT_SPAN_SIZE);
+}
+
+static void az_span_atod_test_no_extra_buffer(void** state)
```

Comment:
What happens if I pass the span source as the conversion buffer?

Reply:
AZ_ERROR_INSUFFICIENT_SPAN_SIZE, conversion buffer needs to be at least +1 greater than source

Reply:
Sorry, let me clarify. I meant, if both span source and conversion buffer are from the same array, just different slices.

This could be a good unit test.
```C
uint8 buffer[10] = { '1', '2', '3' }; // The number 123
az_span source = az_span_slice(AZ_SPAN_FROM_BUFFER(buffer), 0, 4);
az_span conv_span = AZ_SPAN_FROM_BUFFER(AZ_SPAN_FROM_BUFFER(buffer), 0, 5);

double d;
az_result res = az_span_atod(source, &conv_span, &d);
```